### PR TITLE
Update developer guide with the new create-eth workflow

### LINKED ISSUE #6: Updating developer guide
https://github.com/scaffold-eth/create-eth/blob/main/contributors/DEVELOPER-GUIDE.md

With the new workflow on this repo.

### PR DESCRIPTION
**Pull request by Pabl0cks**
Updating the workflow to backmerge `scaffold-eth/scaffold-eth-2:main` branch into `scaffold-eth/create-eth:main` branch.

Based on #12 guidelines

Closes #6

### CODE REVIEW

File: contributors/DEVELOPER-GUIDE.md [hunk 1]
```diff
@@ -55,19 +55,27 @@ Once you're happy with the file contents, you'd need to reverse-engineer the cha
...
-## Back-merging main branch / Publishing to NPM (TODO: Update to main branch workflow)
+## Back-merging main branch / Publishing to NPM
 
-1. Make sure you have the latest changes from `main` branch
-2. Checkout to `cli` branch and create a new branch from it eg: `cli-backmerge`
-3. Checkout to `cli-backmerge` branch and `git merge main`
-4. If there are any conflicts, resolve them and commit the changes
-5. Add changeset by doing `yarn changeset add` follow prompt and commit changes, learn more about changeset [here](https://github.com/scaffold-eth/scaffold-eth-2/blob/cli/CONTRIBUTING.md#changeset)
-6. Push the branch and create a PR against `cli` branch
+1. Sync `upstream-main` branch with SE-2 in the GitHub UI, using "Sync fork" (you can also add the remote branch locally and do it locally)
```
Comment (technophile-04) [inline contributors/DEVELOPER-GUIDE.md:60]: Maybe start with: Switch to `upstream-main` branch and sync it with SE-2 repo from GitHub UI, ...

File: contributors/DEVELOPER-GUIDE.md [hunk 2]
```diff
@@ -55,19 +55,27 @@ Once you're happy with the file contents, you'd need to reverse-engineer the cha
...
-> NOTE: The `cli-backmerge` branch should be merged with "Create a merge commit" option instead of "Squash and merge" option into `cli` branch to preserve the commit history and not needing to create an extra commit directly into `cli` to merge `main` to resolve conflicts.
+   ![gh-web-ui](https://github.com/scaffold-eth/create-eth/assets/55535804/29cd684d-bdd0-42e7-a3c1-2a6e879e1a75)
+
+   </details>
+
+2. Pull `upstream-main` branch locally
+3. Create `backmerge-upstream` branch from `main` branch
+4. Merge `upstream-main` => `backmerge-upstream` branch
+5. If there are any conflicts, resolve them and commit the changes
+   > **Common conflict:** You may need to "accept yours" in the root `package.json` file and add the `next:serve` manually to the `template\base` `package.json` file.
```
Comment (technophile-04) [inline contributors/DEVELOPER-GUIDE.md:71]: Ohh I think, we should also mention that sometimes there will conflicts in root `yarn.lock`  file and we should "accept yours" too or reset those changes to yarn.lock

**Review by technophile-04 — COMMENTED** [review]
Thanks @Pabl0cks !! This is nice!! Added a couple comments 

Also we should update this and point it to core-extension instead of RFC-extension: 

https://github.com/scaffold-eth/create-eth/blob/5a891c32fb1c5484b80d4a8d9c6ecbeb184cbc8c/contributors/DEVELOPER-GUIDE.md?plain=1#L32

File: contributors/DEVELOPER-GUIDE.md [hunk 3]
```diff
@@ -55,19 +55,27 @@ Once you're happy with the file contents, you'd need to reverse-engineer the cha
...
+1. Sync `upstream-main` branch with SE-2 in the GitHub UI, using "Sync fork" (you can also add the remote branch locally and do it locally)
+   <details><summary>Screenshot from GitHub UI</summary>
 
-> NOTE: The `cli-backmerge` branch should be merged with "Create a merge commit" option instead of "Squash and merge" option into `cli` branch to preserve the commit history and not needing to create an extra commit directly into `cli` to merge `main` to resolve conflicts.
+   ![gh-web-ui](https://github.com/scaffold-eth/create-eth/assets/55535804/29cd684d-bdd0-42e7-a3c1-2a6e879e1a75)
+
+   </details>
+
+2. Pull `upstream-main` branch locally
+3. Create `backmerge-upstream` branch from `main` branch
```
[the remark this answers is not in the document]
Comment (carletex) [reply contributors/DEVELOPER-GUIDE.md:68]: Agree!

File: contributors/DEVELOPER-GUIDE.md [hunk 4]
```diff
@@ -55,19 +55,27 @@ Once you're happy with the file contents, you'd need to reverse-engineer the cha
...
+
+> NOTE: The `backmerge-upstream` branch should be merged with "Create a merge commit" option instead of "Squash and merge" option into `main` branch to preserve the commit history.
 
 ### Publishing to NPM
 
-Once the previous PR containing `changeset` is merged to `cli` branch, github bot will automatically create a new PR against `cli` branch containing version increment in `package.json` based on `changeset` and will also update `CHANGELOG.md` with respective `changesets` present.
+Once the previous PR containing `changeset` is merged to `main` branch, github bot will automatically create a new PR against `main` branch containing version increment in `package.json` based on `changeset` and will also update `CHANGELOG.md` with respective `changesets` present.
 
-After this GH bot PR is merged to `cli`. It will auto publish a new release to NPM.
+After this GH bot PR is merged to `main`. It will auto publish a new release to NPM.
```
[the remark this answers is not in the document]
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md:81]: Thank you! Tweaked it a bit, like this:

> TIP: If the published CLI doesn't work and exits unexpectedly, use `npm create eth@latest`. This command provides more verbose output, making it easier to identify the error and reason for failure.

File: contributors/DEVELOPER-GUIDE.md (hunk 1 above)
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md]: Updated!

File: contributors/DEVELOPER-GUIDE.md (hunk 2 above)
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md]: Tweaked a bit the format, like that:

5. If there are any conflicts, resolve them and commit the changes. **Common conflicts:**
   - Sometimes there will be conflicts in the root `yarn.lock` file and we should "accept yours", or reset the changes made to `yarn.lock`.
   - You may also need to "accept yours" in the root `package.json` file and manually add `next:serve` to the `template\base` `package.json` file.

File: contributors/DEVELOPER-GUIDE.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md]: Updated! This is the version I pushed, feel free to do some final tweaks over it 🙌:

2. Pull latest changes from `main` branch locally, and create `backmerge-upstream` branch from it
3. Pull latest changes from `upstream-main` branch locally

File: contributors/DEVELOPER-GUIDE.md [hunk 5]
```diff
@@ -55,19 +55,27 @@ Once you're happy with the file contents, you'd need to reverse-engineer the cha
...
+
+   </details>
+
+2. Pull `upstream-main` branch locally
+3. Create `backmerge-upstream` branch from `main` branch
+4. Merge `upstream-main` => `backmerge-upstream` branch
+5. If there are any conflicts, resolve them and commit the changes
+   > **Common conflict:** You may need to "accept yours" in the root `package.json` file and add the `next:serve` manually to the `template\base` `package.json` file.
+6. Run `yarn changeset add` to create the changeset file (select "patch"). Learn more about changeset [here](https://github.com/scaffold-eth/create-eth/blob/main/CONTRIBUTING.md#changeset)
+7. Create a PR from `backmerge-upstream` against `main` branch. Since this is a fork from SE-2, it automatically points to the SE-2 repo, so you have to manually change it.
```
[the remark this answers is not in the document]
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md]: Updated!

**Comment by Pabl0cks** [comment]
Thanks a lot for your review @technophile-04 and @carletex !! ♥

Just pushed the review tweaks and changed the "RFC-extensions" reference to "core-extensions" 🙌

File: contributors/DEVELOPER-GUIDE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Pabl0cks) [reply contributors/DEVELOPER-GUIDE.md:82]: Thank you! Tweaked text a bit, assumed it was "doesn't" but correct it if I missunderstood pls 🙏:

> TIP: If the published CLI doesn't work and exits unexpectedly, use `npm create eth@latest`. This command provides more verbose output, making it easier to identify the error and reason for failure.

**Review by technophile-04 — APPROVED** [review]
I think this looks great! Thanks Pablo!!

**Pull request merged**